### DISCUSSION TITLE
Plex LXC: Issues with hardware transcoding after restoring LXC to different node

### DISCUSSION THREAD

**Discussion by elospace** [tteck/Proxmox#2951] [Q&A]
I wanted to change the node running my plex server, so I restored (from PBS) to a different node (Intel N100 -> i5-12500t). However, I can't get hardware transcoding to work. I've tried setting permissions of /dev/dri/ using chmod 666, also used chown root:plex but it didn't fix the issue. I want to avoid completely setting up a new Plex LXC from scratch (using tteck).

Something I think may be notable is that on the old node, the iGPU was seen as card0 and thus 226:0, but on the new system the iGPU is card1, and therefore 226:1. I made the appropriate changes in the lxc conf file, shown below. Originally, the 226:0 was 226:1. Is there anywhere else I need to make this change in order for plex to be able to utilize the iGPU?

```
arch: amd64
cores: 7
features: nesting=1
hostname: plexprox
memory: 8192
net0: name=eth0,bridge=vmbr0,gw=192.168.2.254,hwaddr=BC:24:11:FF:66:04,ip=192.168.2.50/24,type=veth
onboot: 1
ostype: ubuntu
rootfs: local-nvme0:104/vm-104-disk-0.raw,size=46G
startup: up=15
swap: 4096
tags: proxmox-helper-scripts
lxc.cgroup2.devices.allow: a
lxc.cap.drop:
lxc.cgroup2.devices.allow: c 188:* rwm
lxc.cgroup2.devices.allow: c 189:* rwm
lxc.mount.entry: /dev/serial/by-id  dev/serial/by-id  none bind,optional,create=dir
lxc.mount.entry: /dev/ttyUSB0       dev/ttyUSB0       none bind,optional,create=file
lxc.mount.entry: /dev/ttyUSB1       dev/ttyUSB1       none bind,optional,create=file
lxc.mount.entry: /dev/ttyACM0       dev/ttyACM0       none bind,optional,create=file
lxc.mount.entry: /dev/ttyACM1       dev/ttyACM1       none bind,optional,create=file
lxc.cgroup2.devices.allow: c 226:1 rwm
lxc.cgroup2.devices.allow: c 226:128 rwm
lxc.cgroup2.devices.allow: c 29:0 rwm
lxc.mount.entry: /dev/fb0 dev/fb0 none bind,optional,create=file
lxc.mount.entry: /dev/dri dev/dri none bind,optional,create=dir
lxc.mount.entry: /dev/dri/renderD128 dev/dri/renderD128 none bind,optional,create=file

````

Relevant info:

/etc/modules (host)

`vfio`
`vfio_iommu_type1`
`vfio_pci`
`vfio_virqfd`

/etc/default/grub (host)

`GRUB_DEFAULT=0`
`GRUB_TIMEOUT=5`
`GRUB_DISTRIBUTOR=`lsb_release -i -s 2> /dev/null || echo Debian``
`GRUB_CMDLINE_LINUX_DEFAULT="quiet initcall_blacklist=sysfb_init"`
`GRUB_CMDLINE_LINUX="quiet intel_iommu=on"`

ls -l /dev/dri (on host)

`drwxr-xr-x 2 root root         80 May 10 05:40 by-path`
`crw-rw---- 1 root video  226,   1 May 10 05:40 card1`
`crw-rw---- 1 root render 226, 128 May 10 05:40 renderD128`

ls -l /dev/dri (on Plex lxc)

`drwxr-xr-x 2 root root         80 May 10 05:40 by-path`
`crw-rw---- 1 root video  226,   1 May 10 05:40 card1`
`crw-rw---- 1 root render 226, 128 May 10 05:40 renderD128`

id plex (on plex LXC)

`uid=999(plex) gid=999(plex) groups=999(plex),44(video),104(render),108(ssl-cert)`

EDIT:

I did a fresh install, and using a privileged LXC created using the script, hardware transcoding still doesn't work. The reason I can't do an unprivileged LXC is because I mount the NFS share in /etc/fstab, something that I cannot seem to do in unprivileged LXCs.

**Comment by tteck** [comment]
Hardware transcoding is broken in kernel `6.8.4-3-pve` you can pin kernel `6.8.4-2-pve` or read this[ post](https://github.com/tteck/Proxmox/discussions/2928) for another solution.

**Reply by elospace** [reply]
After pinning the kernel you mentioned, my issue still persists. I have also tried the solution mentioned in the post, but to no avail.

After disabling tone mapping, it seems that my iGPU is suddenly being utilized which is odd because on my old node tone mapping worked just fine. The old node is running the older kernel version as well. So I guess my issue lies with tone mapping..
